services/scan/src/store.ts

```diff
@@ -1084,6 +1098,8 @@ export class Store {
       JSON.stringify(metadata),
       JSON.stringify(layouts)
     );
+    // cache our layouts so they can be immediately used by the interpreter
+    this.cachedLayouts.push(...layoutsWithImages);
```

Comment:
While it happens that `addHmpbTemplate` is called sequentially with each layout in the same order that `getHmpbTemplates` would load them, nothing actually guarantees that's the case. If the order changed for any reason it would result in interpretation bugs.